subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/LocalComponentDependencyMetadata.java

```diff
@@ -96,7 +96,7 @@ public LocalComponentDependencyMetadata(ComponentIdentifier componentId,
     }
 
     private static List<IvyArtifactName> asImmutable(List<IvyArtifactName> artifactNames) {
-        return artifactNames.isEmpty() ? Collections.emptyList() : ImmutableList.copyOf(artifactNames);
+        return artifactNames.isEmpty() ? Collections.emptyList() : artifactNames instanceof ImmutableList ? artifactNames : ImmutableList.copyOf(artifactNames);
```

Comment:
This is already the case inside `ImmutableList.copyOf` AFAICT

Reply:
Maybe I missed something, but it is called `copyOf` and I think it always creates a copy.

From the implementation:
`ImmutableList<E> list = ((ImmutableCollection<E>) elements).asList();` (which in turn does `toArray()`)
And there is no type check for `ImmutableList`. Maybe the underlying array is not copied (indicated in the Javadoc, but I can't really find that in the code).

Reply:
`((ImmutableCollection<E>) elements).asList();` on a `ImmutableList` does return `this`